src/session/index.js

```diff
@@ -17,8 +17,9 @@ const redisOrInMemory = (options = {}) => {
 const sessionOptions = (userOpts, store, req) => {
   const userCookie = userOpts.cookie || {};
   const cookie = Object.assign({}, {
-    secure: defaultIfUndefined(userCookie.secure, !isTest),
-    expires: defaultIfUndefined(userCookie.expires, false)
+    secure: req.secure,
+    expires: defaultIfUndefined(userCookie.expires, false),
+    domain: req.hostname
```

Comment:
What is the reason for putting the domain here and using req.hostname instead?

Reply:
You need to be careful here. Trusting the requests hostname header allows a request to spoof the host it is being served at, so could open the possibility of man-in-the-middle attacks.

Reply:
true, although we have not choice if we want to test our apps in azure :(